zingo-sync/src/sync.rs

```diff
@@ -127,14 +176,84 @@ fn prepare_next_scan_range(sync_state: &SyncState) -> Option<ScanRange> {
     if let Some((lower_range, higher_range)) = selected_scan_range
         .split_at(selected_scan_range.block_range().start + BlockHeight::from_u32(BATCH_SIZE))
     {
-        scan_ranges.splice(index..=index, vec![lower_range.clone(), higher_range]);
-
-        // TODO: set selected scan range to `ignored` so the same range is not selected twice when multiple tasks call this fn
+        let lower_range_ignored =
```

Comment:
okay, i am understanding ScanPriority::Ignored here means [currently in scan]

Reply:
thats right you are following this really well, so LRZ probably had a slightly different vision for scan ranges but in the way we are using them `ignored` is not used and `scanning` does not exist. for now, ignored means scanning until we maybe PR into LRZ or create our own copy